src/dht_proxy_server.cpp

```diff
@@ -429,22 +429,37 @@ DhtProxyServer::subscribe(const std::shared_ptr<restbed::Session>& session)
                             scheduler_.edit(listener.expireNotifyJob, timeout - proxy::OP_MARGIN);
                             s->yield(restbed::OK);
-                            dht_->get(infoHash,
-                                [this, s](const Sp<Value>& value) {
-                                    if (s->is_closed()) return false;
-                                    // Send values as soon as we get them
-                                    Json::StreamWriterBuilder wbuilder;
-                                    wbuilder["commentStyle"] = "None";
-                                    wbuilder["indentation"] = "";
-                                    auto output = Json::writeString(wbuilder, value->toJson()) + "\n";
-                                    s->yield(output, [](const Sp<restbed::Session>& /*session*/){ });
-                                    return true;
-                                }, [s](bool /*ok* */) {
-                                    // Communication is finished
-                                    if (not s->is_closed()) {
-                                        s->close("{}\n");
-                                    }
-                                });
+
+                            if (root.isMember("previous_values") &&
+                                root["previous_values"].asBool()) {
+                                dht_->get(
+                                    infoHash,
+                                    [this, s](const Sp<Value> &value) {
+                                        if (s->is_closed())
+                                            return false;
+                                        // Send values as soon as we get them
+                                        Json::StreamWriterBuilder wbuilder;
+                                        wbuilder["commentStyle"] = "None";
+                                        wbuilder["indentation"] = "";
+                                        auto output = Json::writeString(
+                                                            wbuilder, value->toJson()) +
+                                                        "\n";
+                                        s->yield(output, [](const Sp<restbed::Session>
+                                                                & /*session*/) {});
+                                        return true;
+                                    },
+                                    [s](bool /*ok* */) {
+                                        // Communication is finished
+                                        if (not s->is_closed()) {
+                                            s->close("{}\n");
+                                        }
+                                    });
+                            } else {
+                                // Communication is finished
+                                if (not s->is_closed()) {
+                                    s->close("{}\n");
```

Comment:
Why not just `s->close();` or `s->close("");` ?

Reply:
Because it will be interpreted as a failure client side and will loop (`reader->parse` will throw, and you will have
```
DHT_LOG.w("Listen closed by the proxy server: %s", e.what());
                        state->ok = false;
```)